utilgo/tlsmanager/tlsmanager.go

```diff
@@ -173,9 +174,18 @@ func (m *tlsManager) WatchServerTLSConf(serverName ClientName, handler func(conf
 		case kclient.SecretChangeTypeAdd:
 			fallthrough
 		case kclient.SecretChangeTypeUpdate:
+			if m.tlsClientCertPEM[serverName] != nil && bytes.Compare(secret.Data[TLSSecretCrtName], m.tlsClientCertPEM[serverName]) == 0 {
```

Comment:
I'm guessing that the certificate is being put into tlsClientCertPEM just for caching/verification purposes. 2 questions - 
1) There wouldn't be a case where the the value is accidentally pulled out because of a key collision, right?
2) Why not just verify m.tlsConf directly?

Reply:
1. Yes, in a single service we should not have client and server certificates for the same service at the same time. So there shouldn't be any collisions.
2. That's would be a great option, but we need to compare PEM strings and I could not find a way to extract PEM from the certificate.